test/src/main/java/org/apache/accumulo/test/compaction/ExternalCompaction_2_IT.java

```diff
@@ -301,6 +305,46 @@ public void testDeleteTableCancelsUserExternalCompaction() throws Exception {
     }
   }
 
+  @Test
+  public void testDeleteTableCancelsSleepingExternalCompaction() throws Exception {
+    getCluster().getClusterControl().startCoordinator(CompactionCoordinator.class);
+    getCluster().getClusterControl().startCompactors(Compactor.class, 1, QUEUE4);
+
+    String table1 = this.getUniqueNames(1)[0];
+    try (AccumuloClient client =
+        Accumulo.newClient().from(getCluster().getClientProperties()).build()) {
+
+      createTable(client, table1, "cs4");
+      TableId tid = getCluster().getServerContext().getTableId(table1);
+      writeData(client, table1);
+
+      // The purpose of this test to ensure the sleeping thread is interrupted. The only way the
+      // compactor can cancel this compaction is if it interrupts the sleeping thread.
+      IteratorSetting iteratorSetting = new IteratorSetting(100, SlowIterator.class);
+      // Sleep so long that the test would timeout if the thread is not interrupted.
+      SlowIterator.setSleepTime(iteratorSetting, 600000);
+      SlowIterator.setSeekSleepTime(iteratorSetting, 600000);
+      SlowIterator.sleepUninterruptibly(iteratorSetting, false);
```

Comment:
Locally tried changing this to true and the test will hang forever because the thread can not be interrupted.

Reply:
We should probably review the system iterators to make sure we are not sleeping uninterruptibly.